pkg/client/dtclient/settings_client.go

```diff
@@ -271,61 +271,71 @@ type match struct {
 	matches constraintMatch
 }
 
-func (d *DynatraceClient) findObjectWithMatchingConstraints(ctx context.Context, source SettingsObject) (*match, error) {
+func (d *DynatraceClient) findObjectWithMatchingConstraints(ctx context.Context, source SettingsObject) (match, bool, error) {
 	constraints, err := d.fetchSchemasConstraints(ctx, source.SchemaId)
 	if err != nil {
-		return nil, fmt.Errorf("unable to get details for schema %q: %w", source.SchemaId, err)
+		return match{}, false, fmt.Errorf("unable to get details for schema %q: %w", source.SchemaId, err)
 	}
 
 	if len(constraints.UniqueProperties) == 0 {
-		return nil, nil
+		return match{}, false, nil
 	}
 
 	objects, err := d.listSettings(ctx, source.SchemaId, ListSettingsOptions{})
 	if err != nil {
-		return nil, fmt.Errorf("unable to get existing settings objects for %q schema: %w", source.SchemaId, err)
+		return match{}, false, fmt.Errorf("unable to get existing settings objects for %q schema: %w", source.SchemaId, err)
 	}
 
-	target, err := findObjectWithSameConstraints(constraints, source, objects)
+	target, found, err := findObjectWithSameConstraints(constraints, source, objects)
 	if err != nil {
-		return nil, err
+		return match{}, false, err
 	}
-	return target, nil
+	return target, found, nil
 }
 
-func findObjectWithSameConstraints(schema SchemaConstraints, source SettingsObject, objects []DownloadSettingsObject) (*match, error) {
+func findObjectWithSameConstraints(schema SchemaConstraints, source SettingsObject, objects []DownloadSettingsObject) (match, bool, error) {
 	candidates := make(map[int]constraintMatch)
 
 	for _, uniqueKeys := range schema.UniqueProperties {
 		for j, o := range objects {
-			match, constraintMatches, err := doObjectsMatchBasedOnUniqueKeys(uniqueKeys, source, o)
+			if o.Scope != source.Scope {
+				continue // settings in different Scopes can't be the same
```

Comment:
I am not sure about this check. If two settings have different scopes (for example, HOST-1 and HOST-2), the same uniqueness check could hit them. Or am I missing something obvious?

Reply:
Um... that is exactly the point of this check? Please read the commit message! (And/or my internal slack message from yesterday.)

Our code currently wrongly assumes that uniqueness constraints apply globally while the Settings framework applies them for a given scope. 
A great example of this - being the one we found it with - are the `builtin:host.process-groups.monitoring-state` settings that are also added as an E2E testcase. The process group in the setting must be unique - but is perfectly valid to define many Settings for the same PG in scope of different hosts.

Reply:
Well, it was also my wrong assumption that the checks are global. Well, learned something :)